ql/src/java/org/apache/hadoop/hive/ql/txn/compactor/Initiator.java

```diff
@@ -182,6 +197,10 @@ public void run() {
     } catch (Throwable t) {
       LOG.error("Caught an exception in the main loop of compactor initiator, exiting " +
           StringUtils.stringifyException(t));
+    } finally {
+      if (compactionExecutor != null) {
+        compactionExecutor.shutdownNow();
```

Comment:
what's the reasoning behind this?

Reply:
Fixed this.

Reply:
@adesh-rao, sorry that was actually a right spot, that's basically end of thread execution